src/python/pants/option/option_types.py

```diff
@@ -92,7 +92,7 @@ def __new__(
         metavar: str | None = None,
         mutually_exclusive_group: str | None = None,
         removal_version: str | None = None,
-        removal_hint: str | None = None,
+        removal_hint: _HelpT | None = None,
```

Comment:
You shouldn't be using the `HelpT` type for not-`help`. Please rename the type (`DynamicString`) or use a new type :slightly_smiling_face:

Reply:
Eh, it is a help string though.